runner_benchmark/taskset.py

```diff
@@ -39,43 +36,47 @@ def replay_requests(self):
         user_wait_time_max = int(os.getenv('USER_WAIT_TIME_MAX_SECONDS', 0))
 
         for request in self.requests:
-            request_base_url = "{0.scheme}://{0.netloc}".format(urlsplit(request['url']))
-            request['url'] = request['url'].replace(request_base_url, self.base_url)
-
             if request['method'] == 'GET':
-                response = self.get(**request)
+                response = self.get(request['url'])
 
                 if response.status_code not in [200, 302]:
                     raise Exception(
                         f"Got a ({response.status_code}) back when getting page: {request['url']}"
                     )
 
-                print("Waiting after GET request")
-
                 if user_wait_time_min and user_wait_time_max:
+                    print("Waiting after GET request")
                     time.sleep(r.randrange(user_wait_time_min, user_wait_time_max))
-            elif request['method'] == 'POST':
 
+            elif request['method'] == 'POST':
                 print('POST to ', request['url'])
 
-                response = self.post(**request)
+                response = self.post(request['url'], request['data'])
 
                 if response.status_code != 302:
-                    raise Exception(f"Got a non-302 ({response.status_code}) back when posting page: {request['url']} with data: {request['data']}")
+                    raise Exception(
+                        f"Got a non-302 ({response.status_code}) back when posting page: {request['url']} with data: {request['data']}"
+                    )
 
                 print(f"Redirect to: {response.headers['Location']}")
 
             else:
-                raise Exception(f"Invalid request method {request['method']} for request to: {request['url']}")
+                raise Exception(
+                    f"Invalid request method {request['method']} for request to: {request['url']}"
+                )
 
     def do_launch_survey(self):
 
         token = create_token(schema_name=self.schema_name)
 
         url = f'/session?token={token}'
-        response = self.get(url=url, name='/session')
+        response = self.get(url)
 
         if response.status_code != 302:
-            raise Exception('Got a non-302 back when authenticating session: {}'.format(response.status_code))
+            raise Exception(
+                'Got a non-302 back when authenticating session: {}'.format(
+                    response.status_code
+                )
+            )
 
-        return self.get(url=response.headers['Location'])
+        response = self.get(response.headers['Location'], allow_redirects=True)
```

Comment:
How come this is assigned to a variable?

Reply:
Originally I was using the response. I've removed the variable.